File: 4/2/cpp/main.cpp
#include <iostream>
#include <iterator>
#include <regex>
#include <string>

std::string swap(std::string text) {
    text = std::regex_replace(text, std::regex("&"), "&amp;");
    text = std::regex_replace(text, std::regex("<"), "&lt;");
    text = std::regex_replace(text, std::regex(">"), "&gt;");
    return text;
}

int main() {
    std::string test_text = "Foo<&>Bar";
    std::cout << "Initial: " << test_text << std::endl; 
    std::cout << "  After: " << swap(test_text) << std::endl; 
}

File: 4/2/rust/src/main.rs
fn main() {
    let start_value = "Foo<&>Bar".to_string();
    println!("Initial: {}", start_value);

    let end_value = swap(start_value);
    println!(" Result: {}", end_value);
}

fn swap(input: String) -> String {
    return input.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;");
}
